tools/src/main/java/org/apache/kafka/tools/consumer/ConsoleShareConsumerOptions.java

```diff
@@ -0,0 +1,268 @@
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.tools.consumer;
+
+import joptsimple.OptionException;
+import joptsimple.OptionSpec;
+import org.apache.kafka.clients.consumer.AcknowledgeType;
+import org.apache.kafka.clients.consumer.ConsumerConfig;
+import org.apache.kafka.common.MessageFormatter;
+import org.apache.kafka.common.utils.Utils;
+import org.apache.kafka.server.util.CommandDefaultOptions;
+import org.apache.kafka.server.util.CommandLineUtils;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Properties;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+public final class ConsoleShareConsumerOptions extends CommandDefaultOptions {
+    private final OptionSpec<String> messageFormatterOpt;
+    private final OptionSpec<String> messageFormatterConfigOpt;
+    private final OptionSpec<String> messageFormatterArgOpt;
+    private final OptionSpec<String> keyDeserializerOpt;
+    private final OptionSpec<String> valueDeserializerOpt;
+    private final OptionSpec<Integer> maxMessagesOpt;
+    private final OptionSpec<?> rejectMessageOnErrorOpt;
+    private final OptionSpec<?> rejectOpt;
+    private final OptionSpec<?> releaseOpt;
+    private final OptionSpec<String> topicOpt;
+    private final OptionSpec<Integer> timeoutMsOpt;
+    private final OptionSpec<String> bootstrapServerOpt;
+    private final OptionSpec<String> groupIdOpt;
+    private final Properties consumerProps;
+    private final MessageFormatter formatter;
+
+    public ConsoleShareConsumerOptions(String[] args) throws IOException {
+        super(args);
+        topicOpt = parser.accepts("topic", "The topic to consume on.")
+                .withRequiredArg()
+                .describedAs("topic")
+                .ofType(String.class);
+        OptionSpec<String> consumerPropertyOpt = parser.accepts("consumer-property", "A mechanism to pass user-defined properties in the form key=value to the consumer.")
+                .withRequiredArg()
+                .describedAs("consumer_prop")
+                .ofType(String.class);
+        OptionSpec<String> consumerConfigOpt = parser.accepts("consumer.config", "Consumer config properties file. Note that " + consumerPropertyOpt + " takes precedence over this config.")
+                .withRequiredArg()
+                .describedAs("config file")
```

Comment:
What's the usage of `describedAs`? I can see sometimes it's `space` separated and sometimes `_`.

Reply:
It's just used in the usage message as a parameter illustration, such as `--topic <String: topic>`.